core/src/main/java/io/grpc/internal/AbstractClientStream.java

```diff
@@ -455,10 +455,10 @@ private void closeListener(
       if (!listenerClosed) {
         listenerClosed = true;
         statsTraceCtx.streamClosed(status);
-        listener().closed(status, rpcProgress, trailers);
         if (getTransportTracer() != null) {
           getTransportTracer().reportStreamClosed(status.isOk());
         }
+        listener().closed(status, rpcProgress, trailers);
```

Comment:
You switched the order of the the listener and transportTracer.
The original tests is about verifying census stats(in statsTraceCtx), how does that related to transportTracer?